lib/db_connection.ex

```diff
@@ -1192,152 +1167,132 @@ defmodule DBConnection do
 
   defp deallocate(conn, query, cursor, opts) do
     conn
-    |> cleanup(&run_deallocate/6, [query, cursor], meter(opts), opts)
+    |> cleanup(&run_deallocate/5, [query, cursor], meter(opts), opts)
     |> log(:deallocate, query, cursor)
   end
 
-  defp run_prepare(conn, conn_state, query, meter, opts) do
-    with {:ok, query, meter} <- prepare(conn, conn_state, query, meter, opts) do
+  defp run_prepare(conn, query, meter, opts) do
+    with {:ok, query, meter} <- prepare(conn, query, meter, opts) do
       describe(conn, query, meter, opts)
     end
   end
 
-  defp prepare(conn, conn_state, query, meter, opts) do
-    conn
-    |> handle(conn_state, :handle_prepare, [query], opts)
-    |> handle_common_result(conn, conn_state, event(meter, :prepare))
+  defp prepare(%DBConnection{pool_ref: pool_ref} = conn, query, meter, opts) do
+    pool_ref
+    |> Holder.handle(:handle_prepare, [query], opts)
+    |> handle_common_result(conn, event(meter, :prepare))
   end
 
-  defp run_prepare_execute(conn, conn_state, query, params, meter, opts) do
-    with {:ok, query, meter}
-          <- run_prepare(conn, conn_state, query, meter, opts),
-         {:ok, params, meter} <- encode(conn, query, params, meter, opts),
-         {:ok, conn_state, meter} <- fetch_info(conn, meter),
```

Comment:
Note we no longer call this `fetch_info`. It is unclear if this `fetch_info` was here to check the status or to get the state. Assuming it was only for the state, then this call is definitely no longer needed.